### DISCUSSION TITLE
None Branded Firmware

### DISCUSSION THREAD

**Discussion by Palingenesis** [winder/Universal-G-Code-Sender#2181] [Ideas]
This is very a nice looking Application. BUT.
It only works with Branded firmware.
May I suggest you add a None Branded option to the list of firmware you can choose from the dropdown box.

Java is not my strong point so I am not able to add it my self.

Having a None Branded choice would enable people who whish to make there own firmware for there machines to use your application.
What do I mean by Unbended:
A choice where your application does not expect a firmware version sent on start up.
The configuration works with the basic "ok" handshaking.
The configuration handles the basic G-Code commands: G00, G01, G02, G03, G04, G91/92..... M03, M04, M05 .......
Does not react to bespoke commands like $, $$.....

Hope you get what I am trying to say, something for those that are experimenting with Arduino.

**Comment by breiler** [comment]
This would mean that you would get a very limited, if any, visual feedback and most buttons would be greyed out. There would be no error handling. 

What features are you expecting us to be able to add?

**Reply by Palingenesis** [reply]
The visual feedback would be the same. You are sending a G-Code File.
It is the same G-code file would send to any of the branded firmware machines.
It's just your Application when you selecting the firmware of your machine from the list, it expects a line of text stating what firmware is on the machine when it connects. "Conformation"
If it does not receive the line of text stating the firmware version, your Application does not continue with the connection.
What I was suggesting was an option that does not require this step.
An option where your application does not require this string.
G-Code is G-Code, BUT there are more machines out there than the ones you have listed in the choices.
I was suggesting a selection that will accept any machine.

As for error handling: That's the point, you need a starting option to find them on a new machine.

Then when someone like myself who has his own machine and firmware would be able to use your software.
Perhaps after finding it works well with your application, may add a version sting at start-up/connection and request it be added to the list.
https://tims-xy-plotter.blogspot.com/2018/10/tims-xy-plotter-this-is-my-latest.html
My machine has custom commands that do more than basic G-Code.
May be I could contribute.
This is My Controller: https://tims-xy-plotter.blogspot.com/2018/12/tims-xy-stepper-controller.html

I do like your graphics.

**Reply by breiler** [reply]
With the little time that I have working on UGS I have to think about getting the biggest bang for the buck. While I like the idea of a lightweight generic protocol implementation. I don't think that the number of users that would benefit from this would outweigh the cost...

I also have a bit of a bad experience with light weight protocols. For instance, I wasted a lot of time trying to implement support for Marlin which resembles a lot how this would have to be implemented. The user experience was too poor and I didn't want that support burden.

**Reply by Palingenesis** [reply]
Oh, I see.
I would not expected support of a none branded firmware option.
It's a shame you don't want an open loop option.
Thank for the prompt replies.
